### PR TITLE
Optimize truncate_stack procedure

### PR DESCRIPTION
**Pull request by Fumuran**
This small PR changes the implementation of the `sys::truncate_stack` procedure in the `stdlib` to make it more efficient, as it was described in the related issue: https://github.com/0xPolygonMiden/miden-vm/issues/1334.

### CODE REVIEW

File: stdlib/asm/sys.masm [hunk 1]
```diff
@@ -4,27 +4,23 @@
...
-    dropw
-    loc_storew.1
-    dropw
-    loc_storew.2
-    dropw
-    loc_storew.3
-    dropw
-    sdepth
-    neq.16
+#! Cycles: 17 + 11 * overflow_words, where `overflow_words` is the number of words needed to drop.
```
Comment (bobbinth) [inline stdlib/asm/sys.masm:7]: nit: I'd add a blank line after line 4 (before "Inputs") and another plank line above line 7 (before "Cycles").

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left one minor nit inline.

**Comment by bobbinth** [comment]
Also - let's update the changelog.

And a question: do we already have good tests covering this procedure?

**Comment by Fumuran** [comment]
In the changelog should I write about change in cycle count and number of locals? 

I think that tests for this procedure are fairly decent: we have both basic test and a proptest [here](https://github.com/0xPolygonMiden/miden-vm/blob/1698d13093c1af367de2d8491940ca9b7137d504/stdlib/tests/sys/mod.rs#L4).

**Comment by bobbinth** [comment]
> In the changelog should I write about change in cycle count and number of locals?

I would just say something like "Optimized `std::sys::truncate_stuck` procedure".

File: stdlib/asm/sys.masm (hunk 1 above)
Comment (bobbinth) [reply stdlib/asm/sys.masm:7]: @Fumuran - could you address this comment as well?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**